Modules/posixmodule.c

```diff
@@ -1782,6 +1788,9 @@ attributes_from_dir(LPCWSTR pszFile, BY_HANDLE_FILE_INFORMATION *info, ULONG *re
     if (n && (pszFile[n - 1] == L'\\' || pszFile[n - 1] == L'/')) {
         // cannot use PyMem_Malloc here because we do not hold the GIL
         filename = (LPCWSTR)malloc((n + 1) * sizeof(filename[0]));
+        if(!filename) {
+            return FALSE;
+        }
```

Comment:
The thread's last error value should be set when the call fails. `malloc()` and WinAPI `HeapAlloc()` don't do this.

```suggestion
        if(!filename) {
            SetLastError(ERROR_NOT_ENOUGH_MEMORY);
            return FALSE;
        }
```

In practice this isn't a problem because `attributes_from_dir()` is only ever called (at least currently) as a fallback in the `stat()` implementation given `CreateFileW()` fails with `ERROR_ACCESS_DENIED` or `ERROR_SHARING_VIOLATION`. But it it's better design to avoid making assumptions about the caller.